### PR TITLE
[v3] Remove support for old ansible versions

### PR DESCRIPTION
**Pull request by gotmax23**
Fixes: https://github.com/ansible-community/antsibull-core/issues/123

### CODE REVIEW

**Comment by felixfontein** [comment]
Failing CI will be fixed by https://github.com/ansible-community/ansible-build-data/pull/353.

**Comment by gotmax23** [comment]
Depends-on: https://github.com/ansible-community/antsibull/pull/571

File: src/antsibull_core/ansible_core.py [hunk 1]
```diff
@@ -189,23 +170,6 @@ async def retrieve(self, ansible_core_version: str, download_dir: StrPath) -> st
         return tar_path
 
 
-def get_ansible_core_package_name(ansible_core_version: str | PypiVer) -> str:
```
Comment (felixfontein) [inline src/antsibull_core/ansible_core.py:192]: This removal needs a changelog fragment.

File: src/antsibull_core/dependency_files.py [hunk 2]
```diff
@@ -58,20 +58,18 @@ def _parse_name_version_spec_file(filename: StrPath) -> DependencyFileData:
     for line in parse_pieces_file(filename):
         record = [entry.strip() for entry in line.split(":", 1)]
 
-        if record[0] in ("_ansible_version", "_acd_version"):
+        if record[0] == "_ansible_version":
```
Comment (felixfontein) [inline src/antsibull_core/dependency_files.py:61]: This should also be mentioned in the changelog.

**Comment by gotmax23** [comment]
I added changelog fragments. Do they look okay?

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@gotmax23 thanks for doing this!

**Pull request merged**